skia/ext/GrGLCreateNativeInterface_glfw.cpp

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2014 Google Inc.
+ *
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+
+
+#include "gl/GrGLInterface.h"
+#include "gl/GrGLAssembleInterface.h"
+#include "gl/GrGLUtil.h"
+
+#include "GLFW/glfw3.h"
+
+static GrGLFuncPtr glfw_get(void* ctx, const char name[]) {
+    SkASSERT(nullptr == ctx);
+    SkASSERT(glfwGetCurrentContext());
+    return glfwGetProcAddress(name);
+}
+
+const GrGLInterface* GrGLCreateNativeInterface() {
+    if (nullptr == glfwGetCurrentContext()) {
+        return nullptr;
```

Comment:
At least on the Mac, you can create a native interface without an active context. Is this not the case on Linux?